keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -254,6 +291,9 @@ If a pod's RestartPolicy is `Never`, the ResizePolicy fields must be set to
 in the container being stopped *and not restarted*, if the system can not
 perform the resize in place.
 
+The `ResizePolicy` field is append-only. Appending the policy is allowed to accommodate a resize
```

Comment:
This field currently has an explicit default, which is why it needs to be mutable. If we remove the explicit defaulting, we could just keep the restriction that it needs to be immutable. My original proposal had this field as fully mutable, but I got pushback based on that being unnecessary (only append is strictly required).

With the removal of the proposed QOS changes from beta scope, we could also reconsider if we need the ability to add and remove resources, since that is only relevant to burstable pods now.

Reply:
Ahh right.  We iterate the resources and populate this.

Am I allowed to take a burstable pod which starts as:

```
requests:
  cpu: 4
  memory: 4Gi
```

and resize it to:

```
requests:
  cpu: 4
```

...which is also burstable ?

If so, I have to be able to remove items from `ResizePolicy`, right?

Reply:
@thockin thanks for your example. In your example, yes, one can resize a burstable Pod in that way.  And yes, that does imply that we need to be able to remove items from ResizePolicy. I think your example shows why the restriction doesn't fully hold. Can we revisit the decision on if ResizePolicy is fully mutable or not?

Reply:
cc/ @mrunalp @derekwaynecarr

Reply:
> If so, I have to be able to remove items from ResizePolicy, right?

No. There isn't any validation that requires resources to be specified for each resource in the resize policy list, so leaving the ResizePolicy behind after removing the memory request is not a problem.

That said, I originally proposed making the `ResizePolicy` list fully mutable. I don't have any problem with relaxing the append-only constraint.

Reply:
Sure you CAN leave it behind, but I think that less "sepcial rules" is better, if we can afford to do that.   I'd be OK to look at that as we move into impl.  If it presents a real problem, we can defer or punt.

Reply:
OK, I updated the KEP to say that the field is fully mutable, but we may reevaluate if corner cases are exposed by the implementation. I think this is what you were suggesting?